pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -703,7 +703,9 @@ public synchronized PinotResourceManagerResponse deleteSegments(String tableName
       Preconditions.checkArgument(TableNameBuilder.isTableResource(tableNameWithType),
           "Table name: %s is not a valid table name with type suffix", tableNameWithType);
       HelixHelper.removeSegmentsFromIdealState(_helixZkManager, tableNameWithType, segmentNames);
-      _segmentDeletionManager.deleteSegments(tableNameWithType, segmentNames);
+      TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableNameWithType);
+      long retentionMs = _segmentDeletionManager.getRetentionMsFromTableConfig(tableConfig);
```

Comment:
Are you trying to get the retention period or deleted segment retention period?

Reply:
this is the deletion segment retention period.

Reply:
I have a question here. why call an API in `SegmentDeletionManager` to get the retention period, and then use that same value to call back another API in `SegmentDeletionManager`? Why not call the deletion API with the (posilby null)  table config and let SegmentDeletionManager figure out whatever is needed from it?

Reply:
actually you are right. with your suggestion earlier to not encode retention Ms when cluster-wide default were used. we no longer need to do this. let me change it.